src/sentry/workflow_engine/endpoints/validators.py

```diff
@@ -0,0 +1,148 @@
+
+    @property
+    def comparison(self) -> Field:
+        raise NotImplementedError
+
+    @property
+    def condition_result(self) -> Field:
+        raise NotImplementedError
+
+    @property
+    def type(self) -> str:
+        # TODO: This should probably at least be an enum
+        raise NotImplementedError
+
+    def validate(self, attrs):
+        attrs = super().validate(attrs)
+        attrs["type"] = self.type
+        return attrs
+
+
+class NumericComparisonConditionValidator(BaseDataConditionValidator):
+
+    comparison = serializers.FloatField(
+        required=True,
+        help_text="Comparison value to be compared against value from data.",
+    )
+    result = serializers.ChoiceField(
+        choices=[
+            (DetectorPriorityLevel.HIGH, "High"),
+            (DetectorPriorityLevel.MEDIUM, "Medium"),
+            (DetectorPriorityLevel.LOW, "Low"),
+        ]
+    )
+
+    @property
+    def supported_conditions(self) -> frozenset[Condition]:
+        raise NotImplementedError
+
+    @property
+    def supported_results(self) -> frozenset[DetectorPriorityLevel]:
+        raise NotImplementedError
+
+    def validate_condition(self, value: str) -> Condition:
+        try:
+            condition = Condition(value)
+        except ValueError:
+            condition = None
+
+        if condition not in self.supported_conditions:
+            raise serializers.ValidationError(f"Unsupported condition {value}")
+        return condition
+
+    def validate_result(self, value: str) -> DetectorPriorityLevel:
+        try:
+            result = DetectorPriorityLevel(int(value))
+        except ValueError:
+            result = None
+        if result not in self.supported_results:
+            raise serializers.ValidationError("Unsupported condition result")
+        return result
+
+
+class BaseDataSourceValidator(CamelSnakeSerializer):
+    pass
+
+
+class BaseGroupTypeDetectorValidator(CamelSnakeSerializer):
+    name = serializers.CharField(
+        required=True,
+        max_length=200,
+        help_text="Name of the uptime monitor",
+    )
+    group_type = serializers.CharField()
+
+    def validate_group_type(self, value: str) -> type[GroupType]:
+        detector_type = grouptype.registry.get_by_slug(value)
+        if detector_type is None:
+            raise serializers.ValidationError("Unknown group type")
+        if detector_type.detector_validator is None:
+            raise serializers.ValidationError("Group type not compatible with detectors")
+        # TODO: Probably need to check a feature flag to decide if a given
+        # org/user is allowed to add a detector
+        return detector_type
+
+    @property
+    def data_source(self) -> BaseDataSourceValidator:
+        raise NotImplementedError
+
+    @property
+    def data_conditions(self) -> BaseDataConditionValidator:
+        raise NotImplementedError
+
+    def create(self, validated_data):
+        # TODO: Implement data source abstraction
```

Comment:
with this in mind, thoughts on `create` calling `create_data_source`, `create_data_conditions` etc (basically decompose this method a bit)

Reply:
I'm not totally sure what the value is - we're always going to create these as a batch in a transaction, and it'll be the same for all detectors - just the data being stored will change.

Reply:
my thought is for code readability and maintenance, it's a bit easier to grok if we reduce complexity.